base_layer/core/src/mempool/service/service.rs

```diff
@@ -161,44 +176,151 @@ impl MempoolService {
             },
         };
 
-        let source_peer_id = msg.propagation_source;
-        let transaction = match Transaction::try_from(msg.message) {
-            Ok(tx) => tx,
-            Err(e) => {
-                warn!(
-                    target: LOG_TARGET,
-                    "Received transaction message from {} with invalid transaction: {:?}", source_peer_id, e
-                );
-                if let Err(err) = self
-                    .network
-                    .report_gossip_message_validation_result(
-                        msg.message_id,
-                        msg.propagation_source,
-                        gossipsub::MessageAcceptance::Reject,
-                    )
-                    .await
-                {
-                    warn!(target: LOG_TARGET, "handle_incoming_tx call network: {err}");
+        let propagation_source = msg.propagation_source;
+        let proto::mempool::NewTransaction { payload: Some(payload) } = msg.message else {
+            warn!(target: LOG_TARGET, "Rejecting gossiped message from peer {propagation_source} because it contains an empty payload.");
+            if let Err(err) = self
+                .network
+                .report_gossip_message_validation_result(
+                    msg.message_id,
+                    propagation_source,
+                    gossipsub::MessageAcceptance::Reject,
+                )
+                .await
+            {
+                warn!(target: LOG_TARGET, "handle_incoming_tx call network: {err}");
+            }
+            return;
+        };
+
+        let transaction = match payload {
+            proto::mempool::new_transaction::Payload::ExcessSig(excess_sig) => {
+                match RistrettoSecretKey::from_canonical_bytes(&excess_sig) {
+                    Ok(excess_sig) => {
+                        let tx_id = MempoolTransactionId::try_from(excess_sig.as_bytes()).unwrap();
+                        match self.inbound_handlers.mempool().has_tx_with_excess_sig(excess_sig).await {
+                            Ok(status) if status.is_new() => {
```

Comment:
dont you need to report [validation](gossipsub::MessageAcceptance::X)  somthing here?
I propose ignore. As I expect us to send a new message out if we added it

Reply:
This is passing it onto the mempool sync protocol for resolving, all messages are resolved there as applicable. E.g if we get the transaction from them, then we Accept, if we get an invalid transaction we Reject